site/src/api/typesGenerated.ts

```diff
@@ -530,6 +545,9 @@ export interface WorkspaceResourceMetadata {
 // From codersdk/workspacebuilds.go
 export type BuildReason = "autostart" | "autostop" | "initiator"
 
+// From codersdk/features.go
+export type Entitlement = "entitled" | "grace_period" | "not_entitled"
```

Comment:
It would be nice if these could be enums

Reply:
Isn't that an enum right now?

Reply:
No, it's a type, which is a little different. Here's the [documentation](https://www.typescriptlang.org/docs/handbook/enums.html) on enums. Basically, enums are key-pairs which are useful because then I can compare API values to desired values on the FE like:
````
{WorkspaceStateEnum[workspaceStatus] !== "Stopped" && (...show this component)}
````
Right now, it looks like the API isn't returning enums at all, which might be a limitation of the generator. If we can get by it, we should. What generator are we using?

Reply:
bikeshed: I prefer string literals since they're closer to JS. Also enums can have some gotchas since depending on how you use them, the value can be a number or a string.

Reply:
Friendship OVER

Reply:
@Kira-Pilot we wrote our own generator :smiley:. So we can do w/e we want.

Reply:
@Kira-Pilot 
https://github.com/coder/coder/blob/7212d7e09ca2b8e58c057d61363d5d3a3381fac7/scripts/apitypings/README.md#L3-L3

Specifically for the enums: https://github.com/coder/coder/blob/7212d7e09ca2b8e58c057d61363d5d3a3381fac7/scripts/apitypings/main.go#L236-L237

(we can refactor this to a go template or something that is easier to read the final format)

Reply:
Sweet! Sounds like maybe @jsjoeio and I can duke it out during FE Variety and we can figure out where to go from there.

Reply:
@Kira-Pilot we have our own generator in `scripts/apitypings`, I believe adding enum support should be fairly trivial, but we would need a way to detect enums in Go code, perhaps an annotation, since Go doesn't have enums.

@jsjoeio there wouldn't be any gotchas with generated code since we'd make sure it's always of the intended type. 💪🏻

Reply:
Oh dang, looks like I'm late to the party /shameful exit.

Reply:
> @jsjoeio there shouldn't be any gotchas with generated code since we'd make sure it's always of the intended type. 💪🏻

Agreed! To clarify, my comment about gotchas wasn't about the generator but rather enums in general and the gotchas around using them in TS.

@matifali you can join the battle in FE Variety and duke it out with us 😉